compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/FlingBehavior.kt

```diff
@@ -27,19 +30,16 @@ import androidx.compose.runtime.Stable
 @Stable
 interface FlingBehavior {
     /**
-     * Perform settling via fling animation with given velocity and suspend until fling has
-     * finished.
-     *
-     * This functions is called with [ScrollScope] to drive the state change of the
-     * [androidx.compose.foundation.gestures.ScrollableState] via [ScrollScope.scrollBy].
-     *
-     * This function must return correct velocity left after it is finished flinging in order to
-     * guarantee proper nested scroll support.
+     * Performs a fling with the given initial velocity.
      *
      * @param initialVelocity velocity available for fling in the orientation specified in
      * [androidx.compose.foundation.gestures.scrollable] that invoked this method.
-     *
+     * @param flingIntoOverscrollEffect The effect containing the logic executed
+     * if the fling inertial motion ends up in overscroll.
      * @return remaining velocity after fling operation has ended
      */
-    suspend fun ScrollScope.performFling(initialVelocity: Float): Float
-}
\ No newline at end of file
+    suspend fun ScrollScope.performFling(initialVelocity: Float, flingIntoOverscrollEffect: FlingIntoOverscrollEffect?): Float
```

Comment:
Ouch. Such things will break bin compatibility with Android. Cannot be merged to 1.5